### PR TITLE
Add optional fields for operations API

### PR DESCRIPTION
**Pull request by ymasiuk**

### CODE REVIEW

File: app/api/v2/management/operations.rb [hunk 1]
```diff
@@ -103,28 +103,27 @@ class Operations < Grape::API
...
             "operations/#{op_type}"
               .camelize
               .constantize
               .order(id: :desc)
               .tap { |q| q.where!(currency_id: currency_id) if currency_id }
               .tap { |q| q.where!(member: member) if member }
-              .page(params[:page])
-              .per(params[:limit])
+              .tap { |q| q.where!(reference_type: reference_type) if reference_type }
+              .tap { |q| q.where!('created_at >= ?', time_from) if time_from.present? }
```
Comment (ysv) [inline app/api/v2/management/operations.rb]: @ymasiuk use 
`Time.at(params[:timestamp])` and remove time_from

File: app/api/v2/management/operations.rb [hunk 2]
```diff
@@ -103,28 +103,27 @@ class Operations < Grape::API
...
 
             "operations/#{op_type}"
               .camelize
               .constantize
               .order(id: :desc)
               .tap { |q| q.where!(currency_id: currency_id) if currency_id }
               .tap { |q| q.where!(member: member) if member }
-              .page(params[:page])
-              .per(params[:limit])
+              .tap { |q| q.where!(reference_type: reference_type) if reference_type }
```
Comment (ysv) [inline app/api/v2/management/operations.rb]: use reference_type from params

File: app/api/v2/management/entities/operation.rb [hunk 3]
```diff
@@ -35,6 +35,11 @@ class Operation < Base
                    type: String,
                    desc: 'The shared user ID.'
                  }) { |operation| operation.try(:member).try(:uid) }
+         expose :reference_type,
```
Comment (ysv) [inline app/api/v2/management/entities/operation.rb]: @ymasiuk use downcased type

File: app/api/v2/management/operations.rb [hunk 4]
```diff
@@ -103,28 +103,27 @@ class Operations < Grape::API
             optional :uid,
                      type: String,
                      desc: 'The user ID for operations filtering.'
-            optional :page,
```
Comment (ysv) [inline app/api/v2/management/operations.rb]: reset pagination

**Review by ysv — CHANGES_REQUESTED** [review]

**Comment by ymasiuk** [comment]
@ysv updated

**Review by ysv — APPROVED** [review]

**Review by mod — COMMENTED** [review]
@calj can you review and explain

File: app/api/v2/management/operations.rb [hunk 5]
```diff
@@ -103,15 +103,27 @@ class Operations < Grape::API
             optional :uid,
                      type: String,
                      desc: 'The user ID for operations filtering.'
+            optional :reference_type,
```
Comment (ysv) [inline app/api/v2/management/operations.rb:120]: @ymasiuk add same params to assets, revenues and expenses

**Comment by ysv** [comment]
@mod 
- There was missing reference type in API we added it
- We added time range filtering to operations API

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/api/v2/management/operations.rb (hunk 5 above)
Comment (ymasiuk) [reply app/api/v2/management/operations.rb:120]: @ysv done

Comment (ysv) [reply app/api/v2/management/operations.rb:120]: @ymasiuk what about time_from & time_to ?

**Review by ysv — CHANGES_REQUESTED** [review]

**Review by ysv — APPROVED** [review]

**Pull request merged**